testing/constraints-3.10.txt

```diff
@@ -0,0 +1,8 @@
+# This constraints file is used to check that lower bounds
+# are correct in setup.py
+# List *all* library dependencies and extras in this file.
+# Pin the version to the lower bound.
+#
+# e.g., if setup.py has "foo >= 1.14.0, < 2.0.0dev",
+# Then this file should have foo==1.14.0
+uritemplate==3.0.1
```

Comment:
Don't we normally pin the minimum version in the constraints file that targets the minimum Python version supported? It would IMO make sense to explain with a comment why this particular constraint lives in the constraints-3.10 file as well (unless this particular library is different).